python/graphstorm/distiller/distiller.py

```diff
@@ -0,0 +1,351 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Distill wrapper.
+"""
+
+import torch as th
+from torch.nn.parallel import DistributedDataParallel
+import os
+import json
+
+from ..model.gnn_distill import GSDistilledModel
+from ..dataloading import DataloaderGenerator, DataManager
+
+class GSdistiller():
+    """ GNN distiller.
+
+    Parameters
+    ----------
+    rank : int
+        The rank.
+        Configs for GNN distillation
+    """
+    def __init__(self, rank):
+        self._rank = rank
+        self._device = -1
+
+    def distill(
+        self, 
+        lm_name, 
+        pre_trained_name,
+        data_dir,
+        batch_size,
+        max_seq_len,
+        distill_lr,
+        saved_path,
+        save_model_frequency,
+        eval_frequency,
+        max_global_step,
+        on_cpu=False,
+    ):
+        """ Distill function.
+        Each worker initializes student model, data provider,
+        and start training.
+
+        Parameters
+        ----------
+        lm_name : str
+            Model name for Transformer-based student model.
+        pre_trained_name : str
+            Name for pre-trained model weights.
+        gnn_embed_dim : int
+            Size of GNN embeddings.
+        data_dir : str
+            Directory for distillation data.
+        batch_size : int
+            Batch size for distillation.
+        max_seq_len : int
+            Maximum sequence length.
+        distill_lr : float
+            Learning rate for distillation.
+        saved_path : str
+            Path to save the model.
+        save_model_frequency : int,
+            Interval to save the model checkpoint.
+        eval_frequency : int,
+            Interval for evaluation.
+        max_global_step : int,
+            Maximum steps for distillation training.
+        on_cpu : bool
+            Whether the distillation will be conducted on cpu.
```

Comment:
do we need this?

Reply:
it's extremely slow to run bert computation on CPUs.

Reply:
let's remove it for now. this is not how trainers enable CPU training. i think we should follow the same convention of trainers.